### ISSUE TITLE
Clearing cache leads to white screens on swipe to go back

### ISSUE THREAD

**Issue by zeeshan-m** [trazek/ionic-modal-nav-example#4]
Whenever 

```
$ionicHistory.clearCache
```

is called within the modal, if I swipe to go back (instead of calling $IonicModalNavService.goBack) the modal just becomes white and the app has to be restarted.

**Comment by zeeshan-m** [comment]
Just realized that this isn't an issue with this library, but it's by design in ionic ([#3317](https://github.com/driftyco/ionic/issues/3317)). Just can't clear cache if you can swipe to go back. It's also worth noting that if you add

```
cache-view="false"
```

to your ion-view, you'll get a white page if you swipe back to get to that page. Essentially, if you disable caching, disable swipe to go back on every page that the caching disabled page can lead to.

**Comment by trazek** [comment]
Understood. Thanks for catching this. I'll reproduce it and see if I can handle that situation

**Comment by zeeshan-m** [comment]
You'll have to reproduce this on a mobile device, I couldn't using ionic serve.

**Comment by trazek** [comment]
Roger that

**Comment by ShawnKuan** [comment]
Hi @zeeshan-m,
Regarding  it's by design in ionic.
I could not understand why the goback button works with disabled cache but swipe back doesn't. 
because the requirement of our project, we need the swipe back feature.
Thanks

**Comment by trazek** [comment]
@zeeshan-m referenced [this issue: #3317](https://github.com/driftyco/ionic/issues/3317). While this seems to be an issue baked into Ionic, I am going to see about incorporating a work around, when I get a chance. I am sorry this is a requirement you may not be able to fulfill using this library, but hopefully caching the back views (for now) is a small compromise you can make.

Thanks for your support!

**Comment by zeeshan-m** [comment]
@ShawnKuan I'm not entirely sure why either, but I believe somewhere in the ionic core the back button causes a re-render of the page when .goBack is called. However, swipe-to-back does not. The way that I had to deal with this was that I just disabled swipe-to-back since disabling caching was more important. Nothing I tried to get the page to reload using the on-swipe-back functions with ionic worked. Sorry :(
